### DISCUSSION TITLE
Does SKTime Forecasting automatically scales amplitudes of the data?

### DISCUSSION THREAD

**Discussion by jobs-git** [alan-turing-institute/sktime#8405] [Q&A]
My dataset has series with different maxes, and many are exponential. Some has maxes of say 1,000,000 whiles others are in 0.0001. But what is important is all of these posses patterns that could appear in any data scale amplitude. If I scale it with the global maximum, we might lose the features of those series with very small maxes. Well, anyway, I already have means to deal with this.

But my question is, does, sktime forecasting already handle normalizing the data already? As I understand, some models need to set encoder length which are shorter than the series length, my assumption is it does normalize by encoder length, so the by this we dont loose the features of those early or late values lows in the series.

But I would also like to confirm this understanding, and what about for data sets with multiple series which has various scale amplitude, does sktime forecasting automatically scales those during training? If its yes, it might not be neccessary to perform this manually (except for other purposes)?

**Comment by fkiraly** [comment]
> But my question is, does, sktime forecasting already handle normalizing the data already?

No, you would need to pipeline with a normalizer, e.g., `StandardScaler` from `scikit-learn` - this should work out of the box.

Some upstream packages, such as `pytorch-forecasting` may do this internally in training, but it is not guaranteed.

**Reply by jobs-git** [reply]
In the TimeSeriesDataset from pytorch-forecasting we can set the scaler right? But I would like to confirm that this scaler is per encoder/decoder_length of a subset of a given series right? not the whole data set?
